### DISCUSSION TITLE
How do I get DeclaringType from a given IObjectInfo

### DISCUSSION THREAD

**Discussion by AlexCrosby** [fluentassertions/fluentassertions#1944] [Q&A] [answered]
I'm refactoring a bunch of code that I have upgraded FluentAssertions from v5 to v6 and have come across an issue where `IMemberInfo` has been replaced with `IObjectInfo` (in this case the info variable).
```cs
.When(info => info != null
           && info.SelectedMemberInfo != null
           && info.SelectedMemberInfo.DeclaringType == foo);
```
I know that in `IMemberInfo` `SelectedInfo` has been flattened down, but `IObjectInfo` does not have a `DeclaringType` and I can see no way to get to it from `IObjectInfo` and I'm not entirely sure whether `CompileTimeType` is equivalent.

**Comment by dennisdoomen** [comment]
That would be `IObjectInfo.Type`.

**Reply by AlexCrosby** [reply]
> That would be `IObjectInfo.Type`.

Cheers, I assumed that would have been the same as `IMemberInfo.type` though? So if `IMemberInfo.DeclaringType == IObjectInfo.Type`, what is `IMemberInfo.Type == IObjectInfo.`?

**Comment by dennisdoomen** [comment]
There is no real difference. They were only different for the root object. But anywhere deeper they were always the same. I don't remember the details. We changed that two years ago.

**Reply by AlexCrosby** [reply]
Sorry back again, unfortunately this seems to not be the solution, all three type variables in `IObjectInfo` seem to refer to the current member type as opposed to the parent type that `SelectedMemberInfo.DeclaringType` used to point to.

Here is a snippet of what I'm talking about:
In the old code the following worked fine when comparing LogMessage objects with a string property called "Message"
```cs
options.Using<string>(ctx => ctx.Subject.Should().Match(ctx.Expectation))
       .When(info => info != null
                  && info.RuntimeType == typeof(string)
                  && info.SelectedMemberInfo.DeclaringType == typeof(LogMessage)
                  && info.SelectedMemberPath.EndsWith("Message"));
```

However with the update it becomes this:
```cs
options.Using<string>(ctx => ctx.Subject.Should().Match(ctx.Expectation))
       .When(info => info != null
                  && info.RuntimeType == typeof(string)
                  && info.Type == typeof(LogMessage)
                  && info.Path.EndsWith("Message"));
```

The issue is that `info.Type`, `RuntimeType` and `CompileTimeType` are all string so the comparison never passes, whereas `IMemberInfo` has the `DeclaringTyp`e which was the Parent (LogMessage).

**Reply by dennisdoomen** [reply]
OK. I need to do some digging.

**Reply by AlexCrosby** [reply]
I assume this feature has been removed, is there an equivalent way of accomplishing the same thing?

**Reply by dennisdoomen** [reply]
No, I'm working on it... https://github.com/fluentassertions/fluentassertions/pull/1950

**Reply by AlexCrosby** [reply]
Thanks for that, much appreciated.

**Comment by jnyrup** [comment, marked as answer]
This is now available in [6.8.0](https://github.com/fluentassertions/fluentassertions/releases/tag/6.8.0).
